zerver/lib/push_notifications.py

```diff
@@ -438,8 +442,11 @@ def add_push_device_token(user_profile, token_str, kind, ios_app_id=None):
                                                                kind=kind,
                                                                ios_app_id=ios_app_id))
     if not created:
+        logging.info("Existing push device updated.")
         token.last_updated = timezone_now()
         token.save(update_fields=['last_updated'])
+    else:
+        logging.info("New push device created.")
```

Comment:
Can we include the user ID in these log lines?

We avoid log statements that are fundamentally per-user activity without some sort of clarifier for which user, since they become useless once several users are involved.

Reply:
In this case, this line is a conclusion that follows the opening at the top of the function -- the opening says all the details of the request including the user ID, and then at each exit point there's a conclusion saying how it turned out.

The options seem to be
* Repeat all the details at each exit point, repetitively.
* Defer the logging until the respective exit points. This is a little more work to write, but more importantly it means we lose the logging if there's some exception in the middle of the logic.
* Leave out the details at exit, and rely on the corresponding entry log to give the details.

I think the last choice is best. In the presence of concurrency it is potentially ambiguous. Right now, though, we don't have enough scale for that to be an issue on this very low-volume path; and before it is an issue we should be adding some kind of request identifier to log messages, to make this and many other sets of related messages more traceable.